Task 03 -  One Piece Website

nortega12 opened the pull request:
# Pull Request
## Description
In this task I developed the One Piece Website from scratch, with mobile first, then tablet and desktop version. Also added the styling of the website and js funtion of the "Read All" functionality.

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
I downloaded the "Live Server" extension in Visual Studio Code so I could see the results while changing the code. You just have to download it, click in "Go Live" in the lower right corner and then it will automatically open a chrome window so you can see the results.

nortega12 committed:
Task 03

html, scss, js of the home page of the one piece website

falvarado3 commented on task03-one-piece-website/onepiece.html:
```diff
@@ -0,0 +1,192 @@
+              <span id="burger-icon" class="icon-times"></span>
+            </button>
+            <div class="logo">
+              <img class="one-piece-logo"src="../images/one-piece-logo.png" alt="onepiece">
```
src doesn't have a space after the class.
Also, give alt a significant name, like "One Piece logo" or some text that represents that image with more clarity

falvarado3 commented on task03-one-piece-website/onepiece.html:
```diff
@@ -0,0 +1,192 @@
+<!DOCTYPE html>
```
Let's fix the indentation on the HTML, which is not consistent throughout the file.

falvarado3 commented on task03-one-piece-website/onepiece.html:
```diff
@@ -0,0 +1,192 @@
+    <div class="main-photo-info-container">
+      <div class="one-piece-container">
+      <img class="img-one-piece-group" src="../images/one-piece-group.png">
+      <img class="img-second-logo" id="img-second-logo" src="../images/one-piece-logo.png">
```
What is the use of this second logo? Is not visible when opening the whole paragraph

nortega12 replied:
it's the One Piece logo that is above the sinopsis container, it appears when you click the "Read All":)

falvarado3 commented on task03-one-piece-website/onepiece.html:
```diff
@@ -0,0 +1,192 @@
+            <a href="/characters" class="see-all-titles">See all</a>
+          </div>
+          <div class="characters-photo-container">
+            <div class="characters-photo-item" id="img-monkey">
+               <p class="characters-photo-text">Monkey D. Luffy</p>
+            </div>
+            <div class="characters-photo-item" id="img-portgas">
+                <p class="characters-photo-text">Portgas D. Ace</p>
+            </div>
+            <div class="characters-photo-item" id="img-roronoa">
+               <p class="characters-photo-text">Roronoa Zoro</p>
+            </div>
+            <div class="characters-photo-item" id="img-nico">
+               <p class="characters-photo-text">Nico Robin</p>
+            </div>
```
There's a lot of repeated code here, I'll give a tip, let's try to add this kind of repetitive code with JS, you can create the objects from there and inject them into the HTML, for example: select the parent container and create the inner HTML of these 4 photo items, if you need more help you can send a message in Slack.

nortega12 replied:
Understood, Pame said that for now I can leave it like that because there's a JS task after, so I can add this tip later in that task, I'll put a reminder when I fix this:)

falvarado3 commented on task03-one-piece-website/onepiece.html:
```diff
@@ -0,0 +1,192 @@
+            <a href="/islands" class="see-all-titles">See all</a>
+          </div>
+          <div class="islands-photo-frames">
+            <div class="islands-photo-item" id="img-goa">
+              <div class="island-text-frame">
+                <p class="island-title">Dawn Island</p>
+                <div class="island-location">
+                  <i class="icon-map-marker"></i>
+                  <p class="island-text">East Blue</p>
+                </div>
+              </div>
+            </div>
+            <div class="islands-photo-item" id="img-reverse">
+              <div class="island-text-frame">
+                <p class="island-title">Reverse Mountain</p>
+                <div class="island-location">
+                  <i class="icon-map-marker"></i>
+                  <p class="island-text">Red Line</p>
+                </div>
+              </div>
+            </div>
+            <div class="islands-photo-item" id="img-reverse">
+              <div class="island-text-frame">
+                <p class="island-title">Reverse Mountain</p>
+                <div class="island-location">
+                  <i class="icon-map-marker"></i>
+                  <p class="island-text">Red Line</p>
+                </div>
+              </div>
```
the same with this ones

falvarado3 commented on task03-one-piece-website/onepiece.html:
```diff
@@ -0,0 +1,192 @@
+            <a href="/mist-objects" class="see-all-titles">See all</a>
+          </div>
+          <div class="objects-photo-container">
+            <div class="objects-photo-item">
+              <div class="object-text-frame">
+                <p class="mist-obj-title">Gomu Gomu No Mi</p>
+                <p class="mist-obj-text">Is a Paramecia-type Devil Fruit that gives the user's body the properties of rubber, making the user a Rubber Human. It was then accidentally eaten by the series protagonist, Monkey D. Luffy.</p>
+              </div>
+            </div>
+            <div class="objects-photo-item">
+              <div class="object-text-frame">
+                <p class="mist-obj-title">Gomu Gomu No Mi</p>
+                <p class="mist-obj-text">Is a Paramecia-type Devil Fruit that gives the user's body the properties of rubber, making the user a Rubber Human. It was then accidentally eaten by the series protagonist, Monkey D. Luffy.</p>
+              </div>
+            </div>
+            <div class="objects-photo-item">
+              <div class="object-text-frame">
+                <p class="mist-obj-title">Gomu Gomu No Mi</p>
+                <p class="mist-obj-text">Is a Paramecia-type Devil Fruit that gives the user's body the properties of rubber, making the user a Rubber Human. It was then accidentally eaten by the series protagonist, Monkey D. Luffy.</p>
+              </div>
+            </div>
+            <div class="objects-photo-item" id="last-obj-item">
+              <div class="object-text-frame">
+                <p class="mist-obj-title">Gomu Gomu No Mi</p>
+                <p class="mist-obj-text">Is a Paramecia-type Devil Fruit that gives the user's body the properties of rubber, making the user a Rubber Human. It was then accidentally eaten by the series protagonist, Monkey D. Luffy.</p>
+              </div>
+            </div>
```
And this ones

falvarado3 commented on task03-one-piece-website/onepiece.js:
```diff
@@ -0,0 +1,26 @@
+ * @summary js with the read all function of the sinopsis section
+ */
+
+function readAll(){
```
Let's start by adding a space after the '()'

falvarado3 commented on task03-one-piece-website/onepiece.js:
```diff
@@ -0,0 +1,26 @@
+    let moreTextBtn = document.getElementById('read-all-btn');
+    let moreTextIcon = document.getElementById('read-all-icon');
+    let secondLogo = document.getElementById('img-second-logo');
+    if(moreText.style.display === "none"){
+        moreText.style.display = "inline";
+        secondLogo.style.display = "block";
+        moreTextBtn.innerHTML = "Read Less";
+        moreTextIcon.className="icon-angle-up";
+
+    }else{
+        moreText.style.display = "none";
+        secondLogo.style.display = "none";
+        moreTextBtn.innerHTML = "Read All";
+        moreTextIcon.className="icon-angle-down";
+    }
```
Respect the spaces after and before the 'if' and 'else' and after the '(......)'

falvarado3 commented on task03-one-piece-website/onepiece.js:
```diff
@@ -0,0 +1,26 @@
+        moreText.style.display = "inline";
+        secondLogo.style.display = "block";
+        moreTextBtn.innerHTML = "Read Less";
+        moreTextIcon.className="icon-angle-up";
```
Add a space after '='

falvarado3 commented on task03-one-piece-website/scss/onepiece.scss:
```diff
@@ -0,0 +1,734 @@
+        display: none;
+    }
+    .icon-align-left {
+        
```
delete this blank line

nortega12 committed:
Fixed

Added spaces between if statements and functions. Also fixed the indentation of the HTML file with shift option f in VS code
